src/Signature/AbstractSignature.php

```diff
@@ -67,9 +74,9 @@ public function normaliseBaseSignatureUrl($url)
     {
         $uri = Uri\UriFactory::factory($url);
         $uri->normalize();
-        if ($uri->getScheme() == 'http' && $uri->getPort() == '80') {
+        if ($uri->getScheme() === 'http' && (string) $uri->getPort() === '80') {
```

Comment:
looking at laminas-uri, the cast is with `(int)`, I think int case is better here for consistency:

```suggestion
        if ($uri->getScheme() === 'http' && (int) $uri->getPort() === 80) {
```